src/async_wrap.cc

```diff
@@ -605,7 +608,16 @@ void AsyncWrap::EmitDestroy(Environment* env, double async_id) {
 // Generalized call for both the constructor and for handles that are pooled
 // and reused over their lifetime. This way a new uid can be assigned when
 // the resource is pulled out of the pool and put back into use.
-void AsyncWrap::AsyncReset(double execution_async_id, bool silent) {
+void AsyncWrap::AsyncReset(double execution_async_id,
+                           bool silent,
+                           bool reused) {
+  if (reused) {
```

Comment:
Would it be possible to initialize the `async_id_` field on the class to e.g. `-1` (or `NaN` if that doesn’t work)? That way we could tell here whether the instance is being re-used without explicitly having to tell it so, right?